lib/AST/PlatformKind.cpp

```diff
@@ -143,14 +143,14 @@ static bool isPlatformActiveForTarget(PlatformKind Platform,
 
 bool swift::isPlatformActive(PlatformKind Platform, const LangOptions &LangOpts,
                              bool ForTargetVariant) {
-  llvm::Triple TT = LangOpts.Target;
+  const llvm::Triple *TT = &LangOpts.Target;
```

Comment:
Just curious if a `const auto &TT = LangOpts.Target;` would also fix the problem but with nicer syntax(not needing the &)? Since ref is just an implicit pointer.

Reply:
I considered that, but I think that would lead to the later `TT = &*LangOpts.TargetVariant;` trying to overwrite the target of the ref, ie, overwriting `LangOpts.Target` itself.

Reply:
... shame though, as it would be much cleaner with what you propose.

Reply:
Since `isPlatformActiveForTarget` takes a ref as well, wonder if something like this:
```cpp
bool swift::isPlatformActive(PlatformKind Platform, const LangOptions &LangOpts,
                              bool ForTargetVariant) {
   if (ForTargetVariant) {
     assert(LangOpts.TargetVariant && "Must have target variant triple");
     return isPlatformActiveForTarget(Platform, *LangOpts.TargetVariant,
                                    LangOpts.EnableAppExtensionRestrictions);
   }

   return isPlatformActiveForTarget(Platform, LangOpts.Target,
                                    LangOpts.EnableAppExtensionRestrictions);
```
looks better? To me it looks cleaner but I know that at the end is just a stylistic discussion, but when reading this pointer `&` and `*`, it is hard to understand.

Reply:
That looks better to me, thanks!  Assuming no-one else gives a strong opinion a different way, i'm happy to change it to the above.

Reply:
Sounds good, I think it is fine as is rather stylistic =]

Reply:
I like the version you suggested, so i've made that change now and tested locally. Will kick off another CI run too.